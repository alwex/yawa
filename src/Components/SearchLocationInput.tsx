import React from 'react'
import { Input, Item } from 'native-base'
import Location from '../Types/Location'
import { geocoder } from '../Services/GeocoderService'

interface SearchLocationInputProps {
  onSearchResult: (locations: Location[]) => void
}
export default class SearchLocationInput extends React.Component<SearchLocationInputProps> {
  async onChange(text: string) {
    let locations: Location[] = []
    const { onSearchResult } = this.props
    if (text.length > 2) {
      locations = await geocoder.searchByName(text)
    }

    onSearchResult(locations)
  }

  render() {
    return (
      <Input placeholder="Search location or city" onChangeText={(text: string) => this.onChange(text)} />
    )
  }
}
