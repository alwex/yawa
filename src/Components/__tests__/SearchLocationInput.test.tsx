import React from 'react'
import renderer from 'react-test-renderer'
import SearchLocationInput from '../SearchLocationInput'

describe('SearchLocationInput', () => {
  it('render correctly', () => {
    const component = renderer.create(<SearchLocationInput searchLocations={() => {}} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
