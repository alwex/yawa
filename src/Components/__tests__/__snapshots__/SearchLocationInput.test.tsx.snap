// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLocationInput render correctly 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  onChangeText={[Function]}
  placeholder="Search location or city"
  placeholderTextColor="#575757"
  style={
    Object {
      "color": "#000",
      "flex": 1,
      "fontSize": 17,
      "height": 50,
      "paddingLeft": 5,
      "paddingRight": 5,
    }
  }
  underlineColorAndroid="rgba(0,0,0,0)"
/>
`;
